### PR TITLE
SAI Release Notes for 1.6.5

### PR DESCRIPTION
**Pull request by kannankvs**

### CODE REVIEW

**Comment by rlhui** [comment]
@DeanKalla, @michaelli10 - would you please help to review/approve this SAI 1.6 release notes, (saimacsec and saisystemport) are two main features added in this release. Thanks.

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+11. saiqosmap
+12. saiqueue
+13. sairouterinterface
+14. saiswitch
+15. saitam
+16. saitunnel
+17. saitypes
+
+
+### sah.h
```
Comment (rlhui) [inline doc/SAI_1.6.5_ReleaseNotes.md:122]: typo here

**Review by rlhui — COMMENTED** [review]
@phshaikh , @bandaru-viswanath , @marian-pritsak , @AshokDaparthi , @JaiOCP , would you please help review/approve this release notes?

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST
+SAI_SWITCH_ATTR_NUMBER_OF_SYSTEM_PORTS
+SAI_SWITCH_ATTR_SYSTEM_PORT_LIST
+SAI_SWITCH_ATTR_NUMBER_OF_FABRIC_PORTS
+SAI_SWITCH_ATTR_FABRIC_PORT_LIST
+
+```
+
+###saitam.h
+Definition for metadata fragment has been enabled.  When there is insufficient space in the packet to add INT metadata for this hop (e.g. MTU would be exceeded), the device may remove the metadata from the packet, send a report to the collector, and insert its metadata before forwarding the packet. SAI_TAM_REPORT_ATTR_ENTERPRISE_NUMBER has been modified for the saitam file.
```
Comment (bandaru-viswanath) [inline doc/SAI_1.6.5_ReleaseNotes.md:467]: SAI_TAM_REPORT_ATTR_ENTERPRISE_NUMBER has been added as an attribute for the TAM Report object in the saitam.h file.

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+     */
+    SAI_TAM_REPORT_ATTR_ENTERPRISE_NUMBER,
+
+    /**
+	
+```
+
+
+### saitunnel.h	
+Create and Set for Tunnel Attributes are added.
```
Comment (bandaru-viswanath) [inline doc/SAI_1.6.5_ReleaseNotes.md:501]: Support for P2P tunnel types added, in addition to existing P2MP mode. Additionally, some attributes are now allowed to be set after creation of the tunnel objects.

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+     * @brief P2P Tunnel
+     */
+    SAI_TUNNEL_PEER_MODE_P2P,
+
+    /**
+     * @brief P2MP Tunnel
+     */
+    SAI_TUNNEL_PEER_MODE_P2MP,
+
+} sai_tunnel_peer_mode_t;
```
Comment (bandaru-viswanath) [inline doc/SAI_1.6.5_ReleaseNotes.md:519]: I think the sai_tunnel.h has more diffs, than what is presented here.

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 5]
```diff
@@ -0,0 +1,539 @@
...
+1)Attribute data for #SAI_PORT_ATTR_INTERFACE_TYPE. Used for selecting electrical interface with specific electrical pin and signal quality. <br>
+2)Port bind point for ingress/egress ACL object. Bind (or unbind) an ingress/egress ACL table or ACL group on a port. Enable/Update ingress/egress ACL table or ACL group filtering by assigning the list of valid object id. Disable ingress/egress filtering by assigning SAI_NULL_OBJECT_ID in the attribute value. <br>
+3)Port bind point for ingress MACsec ACL object. Bind (or unbind) an ingress MACsec ACL table on a port. Enable/Update ingress MACsec ACL table filtering by assigning the list of valid object id. Disable ingress filtering by assigning SAI_NULL_OBJECT_ID in the attribute value. <br>
+4)Link training failure status and error codes port stat PRBS error counts and list of port connector attributes are added. <br>
+
+```
+SAI_PORT_ATTR_INGRESS_MACSEC_ACL
+SAI_PORT_ATTR_EGRESS_MACSEC_ACL
+SAI_PORT_ATTR_MACSEC_PORT_LIST
+
```
Comment (michaelli10) [inline doc/SAI_1.6.5_ReleaseNotes.md:372]: Will you add the system port changes (and other saiport.h changes) from PR1080?https://github.com/opencomputeproject/SAI/pull/1081

    /**
     * @brief System port for the port
     *
     * @type sai_object_id_t
     * @flags READ_ONLY
     * @objects SAI_OBJECT_TYPE_SYSTEM_PORT
     */
    SAI_PORT_ATTR_SYSTEM_PORT,

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 6]
```diff
@@ -0,0 +1,539 @@
...
+4)Driver will give this context back to adapter as part of call back sai_switch_register_read/write_fn API. <br>
+5)Platform adaption device write callback function passed to the adapter. This is mandatory function for driver when device access not supported by file system. <br>
+6)Platform specific device register read access. This API provides platform adaption functionality to access device registers from driver. This is mandatory to pass as attribute to sai_create_switch when driver implementation does not support register access by device file system directly.<br>
+7)Platform specific device register write access. This API provides platform adaption functionality to access device registers from driver. This is mandatory to pass as attribute to sai_create_switch when driver implementation does not support register access by device file system directly.<br>
+8)Switch MDIO read API - Provides read access API for devices connected to MDIO from NPU SAI.<br>
+9)Switch MDIO write API - Provides write access API for devices connected to MDIO from NPU SAI.<br>
+
```
Comment (michaelli10) [inline doc/SAI_1.6.5_ReleaseNotes.md:428]: Can you add an item about the new system port related attributes?

SAI_SWITCH_ATTR_SWITCH_ID
SAI_SWITCH_ATTR_MAX_SYSTEM_CORES
SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST
SAI_SWITCH_ATTR_NUMBER_OF_SYSTEM_PORTS
SAI_SWITCH_ATTR_SYSTEM_PORT_LIST
SAI_SWITCH_ATTR_NUMBER_OF_FABRIC_PORTS
SAI_SWITCH_ATTR_FABRIC_PORT_LIST

**Review by michaelli10 — COMMENTED** [review]
Does it make sense to separate the changes by feature similar to SAI 1.5 release notes for context?  Also, adding the PR references would be helpful.  https://github.com/opencomputeproject/SAI/blob/v1.6/doc/SAI_1.5_ReleaseNotes.md

**Comment by DeanKalla** [comment]
> @DeanKalla, @michaelli10 - would you please help to review/approve this SAI 1.6 release notes, (saimacsec and saisystemport) are two main features added in this release. Thanks.

saimacsec changes look good to me

**Comment by rlhui** [comment]
> > @DeanKalla, @michaelli10 - would you please help to review/approve this SAI 1.6 release notes, (saimacsec and saisystemport) are two main features added in this release. Thanks.
> 
> saimacsec changes look good to me

@DeanKalla - thanks. would you please approve if it's good to you?

File: doc/SAI_1.6.5_ReleaseNotes.md (hunk 4 above)
Comment (kannankvs) [reply doc/SAI_1.6.5_ReleaseNotes.md:519]: Done

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 7]
```diff
@@ -0,0 +1,539 @@
...
+    sai_get_macsec_sa_attribute_fn      get_macsec_sa_attribute;
+    sai_get_macsec_sa_stats_fn          get_macsec_sa_stats;
+    sai_get_macsec_sa_stats_ext_fn      get_macsec_sa_stats_ext;
+    sai_clear_macsec_sa_stats_fn        clear_macsec_sa_stats;
+
+```
+
+### saisystemport.h
+
+System port creation can be accomplished by calling switch_create() with the list of system ports to be created using SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST or creating them individually using the create_system_port() function in the system port APIs. Although it is expected that all local ports be created in switch_create().
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SAI_1.6.5_ReleaseNotes.md]: Done

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 8]
```diff
@@ -0,0 +1,539 @@
...
+    sai_get_fine_grained_hash_field_attribute_fn   get_fine_grained_hash_field_attribute;
+	
+```	
+
+### saihostif.h
+Definition type for default action on IPv6 neighbor solicitation, trap types for mpls, and egress queue index have been modified for the saihostif file.
+
+```
+
+ /** Default packet action is forward */
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SAI_1.6.5_ReleaseNotes.md]: Done

File: doc/SAI_1.6.5_ReleaseNotes.md [hunk 9]
```diff
@@ -0,0 +1,539 @@
...
+
+```
+    * @brief TPID
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @isvlan false
+     * @default 0x8100
+     */
+    SAI_LAG_ATTR_TPID,
```
[the remark this answers is not in the document]
Comment (kannankvs) [reply doc/SAI_1.6.5_ReleaseNotes.md:277]: Yes, as its included in the new version

**Comment by rlhui** [comment]
> Does it make sense to separate the changes by feature similar to SAI 1.5 release notes for context? Also, adding the PR references would be helpful. https://github.com/opencomputeproject/SAI/blob/v1.6/doc/SAI_1.5_ReleaseNotes.md

@kannankvs  - yes, please add that the below features can be supported with this SAI release:

-	Gearbox/Phy/MACsec Support
-	VOQ System
-	MPLS Segment Routing Enhancements
-	Fine-grained and ordered ECMP types
-	ACL GRE key match
-	ACL support to match TAM/INT packets
-	Bulk api for fdb entry
-	NRZ and PAM-4 in meta dictionary
-	Enable single chip data plane testing without external TG
-	TPID configuration
-	IPFix enhancement
-	Add support for matching any INT packets
-	Metadata enhancements

**Comment by rlhui** [comment]
@michaelli10  - please check if your comments are addressed? Thanks.

**Comment by rlhui** [comment]
@phshaikh , @marian-pritsak , @AshokDaparthi , @JaiOCP  - would you please help to review/sign off this RN? Thanks.

**Comment by kannankvs** [comment]
SONiC 202006 is using SAI 1.6.3 and hence we shall use only the 1.6.3 release notes. We have addressed all the comments given in this PR in the 1.6.3 release notes. Please refer [PR1131](https://github.com/opencomputeproject/SAI/pull/1131) and sign off in that PR1131. We are closing this PR1134 with no further action.

**Pull request closed without merging**